### PR TITLE
OCPBUGS-11137: Add configmap to store subscriber info as a storage type - To prevent loss of subscribers

### PR DESCRIPTION
**Pull request by aneeshkp**
HTTP Publisher needs to persist subscriber in case or pod restarts the subscriber informations are not lost.
This requires permission to create configmap 
configmap are named per node. So it is necessary to set namespace and codename in the environment variable.
Consumer doesn't require any persistence of publisher  info

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -74,16 +72,9 @@ require (
...
-	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
+	k8s.io/klog/v2 v2.80.1 // indirect
+	k8s.io/kube-openapi v0.0.0-20221012153701-172d655c2280 // indirect
+	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
-
-// Manually pinned to kubernetes-1.24.0
-replace (
```
Comment (nishant-parekh) [inline go.mod:85]: should this be deleted?

File: hack/test-go.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+echo "Running golang staticcheck ..."
+${BASEDIR}/bin/staticcheck --tags=test ./...
+
+echo "Running go tests..."
+KUBEBUILDER_ASSETS="$(pwd)/bin" ${GO} test \
+    --tags=test \
+    -v \
+    -covermode=count \
+    -coverprofile=coverage.out \
+    $(${GO} list ./... | grep -v e2e | tr "\n" " ")
```
Comment (nishant-parekh) [inline hack/test-go.sh:25]: missing new line

**Review by nishant-parekh — COMMENTED** [review]
added some suggestions

File: examples/consumer.Dockerfile [hunk 3]
```diff
@@ -9,7 +9,7 @@ COPY . .
 
 RUN hack/build-example-go.sh
 
-FROM openshift/origin-base AS bin
+FROM registry.ci.openshift.org/ocp/4.13:base AS bin
```
[the remark this answers is not in the document]
Comment (aneeshkp) [reply examples/consumer.Dockerfile:12]: that's consumer

File: Makefile [hunk 4]
```diff
@@ -124,3 +124,10 @@ fmt-code: ## Run go fmt against code.
 
 vet: ## Run go vet against code.
 	go vet ./...
+
+generate-api:
+	hack/verify-codegen.sh
```
Comment (nishant-parekh) [inline Makefile:129]: the script this is running doesnt exist, same for install-tools below.. should these be cleaned up as well?

**Review by nishant-parekh — COMMENTED** [review]
looks good just one small comment

**Review by nishant-parekh — APPROVED** [review]

**Pull request merged**